contrib/rtlbrowse/logfile.c

```diff
@@ -93,1390 +89,1308 @@ static GtkTextTag *blue_tag = NULL, *fwht_tag = NULL;
-				{
-				if(!is_identifier(sel[i])) goto bail;
-				}
-
-			while(gtk_text_iter_backward_char(&start))
-				{
-		               	sel2 = gtk_text_buffer_get_text(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-                                              &start, &end, FALSE);
-				if(!sel2) break;
-				ch = *sel2;
-				g_free(sel2);
-				if(!is_identifier(ch))
-					{
-					gtk_text_iter_forward_char(&start);
-					break;
-					}
-				}
-
-			gtk_text_iter_backward_char(&end);
-			for(;;)
-				{
-				gtk_text_iter_forward_char(&end);
-		               	sel2 = gtk_text_buffer_get_text(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-                                              	&start, &end, FALSE);
-				if(!sel2) break;
-				ch = *(sel2 + strlen(sel2) - 1);
-				g_free(sel2);
-				if(!is_identifier(ch))
-					{
-					gtk_text_iter_backward_char(&end);
-					break;
-					}
-				}
-
-		 	sel2 = gtk_text_buffer_get_text(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
-						&start, &end, FALSE);
-
-
-			/* oec->set_selection(oe, lft, rgh); */
-
-			import_doubleclick(text, sel2);
-			g_free(sel2);
-			}
-bail:			g_free(sel);
-			}
-		}
-	}
+    while (ft) {
+        if (!strcmp(s2, ft->fullname)) {
+            bwlogbox(ft->fullname, 640 + 8 * 8, ft, 0);
+            break;
+        }
 
+        ft = ft->next_flat;
+    }
 
-return(FALSE); /* call remaining handlers... */
+    free(s2);
 }
 
-static        gint
-scroll_event( GtkWidget * widget, GdkEventScroll * event, gpointer text)
+// Decide whether the text being double-clicked is a part of an valid
+// verilog identifier. If so, call `import_doubleclick`
+static void button_release_event(GtkGestureClick *gesture,
+            guint            n_pressed,
+            double           x,
+            double           y,
+            gpointer         text)
 {
-(void)widget;
-
-  GtkTextView *text_view = GTK_TEXT_VIEW(text);
-  /* GtkAdjustment *hadj = text_view->hadjustment; */
-  GtkAdjustment *vadj = YYY_gtk_text_view_get_vadjustment(YYY_GTK_TEXT_VIEW(text_view));
-
-  gdouble s_val = gtk_adjustment_get_step_increment(vadj);
-  gdouble p_val = gtk_adjustment_get_page_increment(vadj);
-
-  switch ( event->direction )
-  {
-    case GDK_SCROLL_UP:
-	gtk_adjustment_set_value(vadj, gtk_adjustment_get_value(vadj) - s_val);
-	if(gtk_adjustment_get_value(vadj) < gtk_adjustment_get_lower(vadj)) gtk_adjustment_set_value(vadj, gtk_adjustment_get_lower(vadj));
-
-	g_signal_emit_by_name (GTK_ADJUSTMENT(vadj), "changed");
-        g_signal_emit_by_name (GTK_ADJUSTMENT(vadj), "value_changed");
-
-      break;
+    (void)gesture;
+    (void)n_pressed;
+    (void)x; (void)y;
+    gchar *sel;
+
+    GtkTextIter start;
+    GtkTextIter end;
+
+    if (gtk_text_buffer_get_selection_bounds(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
+                                             &start,
+                                             &end)) {
+        if (gtk_text_iter_compare(&start, &end) < 0) {
```

Comment:
This is another case where the nesting could be reduced to improve readability:
```suggestion
    if (!gtk_text_buffer_get_selection_bounds(gtk_text_view_get_buffer(GTK_TEXT_VIEW(text)),
                                             &start,
                                             &end)) {
        return;
    }

    if (gtk_text_iter_compare(&start, &end) >= 0) {
        return;    
    }
```